### ISSUE TITLE
Lint againts items with the same names as their modules

### ISSUE THREAD

**Issue by SLUCHABLUB** [rust-lang/rust-clippy#14226]
### What it does

In the same vein as how the `module_name_repetitions` lint warn against the likes of `foo:FooBar`, I believe that there should be a lint that warns against `foo:Foo`. The rationale behind the `module_name_repetitions` lint is that you are forced to write the module name twice, which you also are in the case that the item name _is_ the module name. Note that the `module_name_repetitions` lint only checks for if the item names that are _"prefixed or suffixed"_ by their module name.

### Advantage

- Makes use statements shorter and more readable.
- Removes the need to write the module name twice.

### Drawbacks

- Does not follow the same style as `std` for example: `std::error::Error`

### Example

```rust
mod bar {
    pub mod foo {
        pub struct Foo;
    }
}

use bar::foo::Foo;
```

Could be written as:

```rust
mod bar {
    mod foo {
        pub struct Foo;
    }

    pub use foo::Foo;
}

use bar::Foo;
```
